docs/getting_started/integration_tests.md

```diff
@@ -0,0 +1,164 @@
+
+This package acts as a wrapper over Golang's _testing_, and each _Suite_ object implements the following interface:
+
+[embedmd]:# (../../pkg/test/suite/suite.go /type TestingSuite interface/ /\n}/)
+```go
+type TestingSuite interface {
+	Env() *env.Environment
+	SetEnv(environment *env.Environment)
+	SetT(t *testing.T)
+	T() *testing.T
+	SetupSuite() []Option
+}
+```
+
+The starting point for a Gosoline integration test execution is the `suite.Run` function:
+
+[embedmd]:# (../../pkg/test/suite/run.go /func Run/ /\)/)
+```go
+func Run(t *testing.T, suite TestingSuite, extraOptions ...Option)
+```
+
+This method will use reflection to find all test cases declared by a given _TestingSuite_ object, apply the Option then create a kernel with whatever modules or APIs were declared in the `SetupSuite` and `SetupApiDefinitions`, and lastly run the tests.
+
+Each Gosoline integration test follows the same format:
+
+- Creates an object which implements _TestingSuite_
+- Implements the `SetupSuite` method for that object
+- Has at least one `Test...` method
+- It calls `suite.Run`
+
+## Practical example
+
+Inside `examples/getting_started/integration` we have an integration test for _money-exchange_. It is comprised out of two files, `api_test.go` and `fixtures.go`.
+
+`fixtures.go` contains initial values for the `currency` key value store. Its most important part is:
+
+[embedmd]:# (../../examples/getting_started/integration/fixtures.go /var fixtureSets/ /\n}/)
+```go
+var fixtureSets = []*gosoFixtures.FixtureSet{
+	{
+		Enabled: true,
+		Writer:  gosoFixtures.ConfigurableKvStoreFixtureWriterFactory("currency"),
+		Fixtures: []interface{}{
+			&gosoFixtures.KvStoreFixture{
+				Key:   "GBP",
+				Value: 1.25,
+			},
+			&gosoFixtures.KvStoreFixture{
+				Key:   "2021-01-03-GBP",
+				Value: 0.8,
+			},
+		},
+	},
+}
+```
+
+This _FixtureSet_ holds exchange rate data for GBP, which will be needed by the test. We use fixtures to predefine test data, because we do not want to download actual exchange rates just for one test. 
+
+`api_test.go` declares _ApiTestSuite_ which will implement _TestingSuite_ :
+
+[embedmd]:# (../../examples/getting_started/integration/api_test.go /type ApiTestSuite/ /\n}/)
+```go
+type ApiTestSuite struct {
+	suite.Suite
+
+	clock clock.Clock
+}
+```
+
+It also declares a normal unit test which makes use of the above struct, and calling _suite.Run_:
+
+[embedmd]:# (../../examples/getting_started/integration/api_test.go /func TestApiTestSuite/ /\n}/)
+```go
+func TestApiTestSuite(t *testing.T) {
+	suite.Run(t, &ApiTestSuite{
+		clock: clock.NewFakeClockAt(time.Now().UTC()),
+	})
+}
+```
+
+Notice the use of `clock.NewFakeClockAt`. This is an elegant solution to the following problem: when testing the same code multiple times, we want the test results to be identical. For code which makes calls to `time.Time()` this is by default not true. Using a fake clock which always returns a predefined time, solves this problem. 
```

Comment:
But your code also uses `time.Now()` (there is no `time.Time()`), so how does this now work? I would even argue that this API can work perfectly without a fake clock (because you get the rates for the current day, but we always have the current day right now, you are not defining what this day is, and some fixed historical date, which also always stays the same)